druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java

```diff
@@ -1102,24 +1107,46 @@ public String getQueryString(String pagingIdentifier, int offset) {
         }
       }
       this.timeZone = timeZone;
+      this.dateFormatter = new SimpleDateFormat(TimeExtractionFunction.ISO_TIME_FORMAT,
+          Locale.ROOT);
+      if (timeZone != null) {
+        this.dateFormatter.setTimeZone(TimeZone.getTimeZone(timeZone));
+      }
     }
 
     protected void clearFieldNameLists() {
       dimensions.clear();
       metrics.clear();
     }
 
-    @SuppressWarnings("incomplete-switch") String translate(RexNode e, boolean set) {
+    @SuppressWarnings("incomplete-switch")
+    String translate(RexNode e, boolean set, boolean formatDateString) {
       int index = -1;
       switch (e.getKind()) {
       case INPUT_REF:
         final RexInputRef ref = (RexInputRef) e;
         index = ref.getIndex();
         break;
       case CAST:
-        return tr(e, 0, set);
+        return tr(e, 0, set, formatDateString);
       case LITERAL:
-        return ((RexLiteral) e).getValue3().toString();
+
+        if (!formatDateString) {
+          return ((RexLiteral) e).getValue3().toString();
```

Comment:
TBH am always puzzled about the usage of `getValue3` and `getValue`, @jcamachor can you please provide guidance? looking at the code seems like `getValue` is what we want.

Reply:
I think as this might be a timestamp, we should not go through ```getValue``` which would return a Calendar object (then toString would shift the timestamp, etc). I think we could go with ```getValue3```, although I am doubting whether it is valid for every type. Maybe something to explore in a follow-up.

What @b-slim mentioned below about the ```null``` value applies here too. However, note that in SQL ```x=null``` is not equal to ```x is null```. I think we will only hit an Exception for the first expression (and if we do not, it is probably because Calcite folds the expression before reaching the Druid adapter). In any case, I think we can tackle that in follow-up.

Reply:
for getValue3 would need to add explicit cast tried to avoid that with getValueAs and also handle Date case explicitly, have modified the logic and handled null being returned. please check again.